lib/p2p/Peer.ts

```diff
@@ -549,6 +569,113 @@ class Peer extends EventEmitter {
     this.sendPong(packet.header.id);
   }
 
+  private handleDealRequest =  async (request: packets.DealRequest)  => {
+    this.logger.debug('Got this: ' + JSON.stringify(request));
+    const requestBody = request.body;
+    if (!requestBody) {
+      return;
+    }
+    const preImage = crypto.randomBytes(32).toString('hex');
+    const hash = crypto.createHash('sha256');
+    const r_hash = hash.update(preImage).digest('hex');
+    const makerDealId = crypto.randomBytes(32).toString('hex');
+    let makerPubKey:string;
+
+    switch (requestBody.makerCoin){
+      case CoinType.BTC:
+        const infoBtc = await this.pool.lndBtcClient.getInfo();
+        makerPubKey = infoBtc.identityPubkey;
+        break;
+      case CoinType.LTC:
+        const infoLtc = await this.pool.lndLtcClient.getInfo();
+        makerPubKey = infoLtc.identityPubkey;
+        break;
+      default:
+        return;
+    }
+
+    const deal:SwapDeal = {
+      preImage,
+      r_hash,
+      makerDealId,
+      makerPubKey,
+      myRole : SwapDealRole.Maker,
+      takerAmount : requestBody.takerAmount,
+      takerCoin : requestBody.takerCoin,
+      takerPubKey: requestBody.takerPubKey,
+      takerDealId: requestBody.takerDealId,
+      makerAmount: requestBody.makerAmount,
+      makerCoin: requestBody.makerCoin,
+      createTime: Date.now(),
+    };
+
+    const body:packets.DealResponsePacketBody = {
+      makerPubKey,
+      r_hash,
+      makerDealId,
+      takerDealId: requestBody.takerDealId,
+    };
+
+    this.pool.swapDeals.addSwapDeal(deal);
+    this.logger.debug(' swap deal: ' + JSON.stringify(deal));
+
+    this.logger.debug('sending back to peer: '  + JSON.stringify(body));
+
+    const packet = new packets.DealResponse(body, request.header.id);
+
+    this.sendPacket(packet);
+  }
+
+  private handleDealResponse = (response: packets.DealResponse): void  => {
+    if (!response.body) {
+      return;
+    }
+    const deal = this.pool.swapDeals.get(SwapDealRole.Taker, response.body.takerDealId);
+    if (!deal) {
+      return;
+    }
+
+    deal.makerPubKey = response.body.makerPubKey;
+    deal.makerDealId = response.body.makerDealId;
+    deal.r_hash = response.body.r_hash;
+    this.logger.debug(' updated deal: ' + JSON.stringify(this.pool.swapDeals.get(SwapDealRole.Taker, response.body.takerDealId)));
+
+    const body:packets.SwapRequestPacketBody = {
+      makerDealId: deal.makerDealId,
+    };
+    const packet = new packets.SwapRequest(body);
+
+    this.sendPacket(packet);
+  }
+
+  private handleSwapRequest = (request: packets.SwapRequest): void  => {
+    if (!request.body) {
+      return;
+    }
+    const deal = this.pool.swapDeals.get(SwapDealRole.Maker, request.body.makerDealId);
+    if (!deal) {
+      return;
+    }
+    const body:packets.SwapResponsePacketBody = {
+      r_preimage: deal.preImage,
+    };
+    const packet = new packets.SwapResponse(body, request.header.id);
```

Comment:
Before this line is where we'd attempt to perform the actual swap I presume?

Reply:
Yep!